test/e2e/data/infrastructure-docker/v1.2/bases/cluster-with-kcp.yaml

```diff
@@ -0,0 +1,95 @@
+---
```

Comment:
Not sure if the v1.2 folder is a good idea.

The problem is that the resources in the v1beta1 folder contain fields which are only supported on main / v1.3.x
(nodeDeletionTimeout & rolloutBefore).

Reply:
what about having a subfolder called main under v1beta1 instead?

also as a general note, we should start adding some more comments to our yaml documenting why they are there / what is special in them (e.g what are the settings that are not supported in main) / when we can drop them

Reply:
Done